### DISCUSSION TITLE
sha-256-rsa-mgf1 as SignatureMethod?

### DISCUSSION THREAD

**Discussion by cornzz** [node-saml/node-saml#383] [Q&A]
Our IDP is switching from `rsa-sha256` to `sha-256-rsa-mgf1` as the signature method and I was wondering if `node-saml` supports that already, it seems from the following code snippet that it doesn't?

https://github.com/node-saml/node-saml/blob/0d3d3fae7328ca6a7aaaf36dfe030b8f1baf7da4/src/algorithms.ts#L3-L13

What would have to be done to implement support for that signing method?

Also, our IDP is switching the EncryptionMethod from `aes128-cbc` to `aes128-gcm`, but I suppose that doesn't require action on our side, right?

**Comment by srd90** [comment]
>What would have to be done to implement support for that signing method?

See e.g.:
* https://github.com/node-saml/xml-crypto/issues/328
